src/wazuh_modules/syscollector/src/syscollectorImp.cpp

```diff
@@ -1626,6 +1634,22 @@ void Syscollector::sync()
     m_logFunction(LOG_DEBUG, "Ending syscollector sync");
 }
 
+void Syscollector::syncAlgorithm()
+{
+    m_currentIntervalValue = m_intervalValue / 2 >= MAX_DELAY_TIME ? MAX_DELAY_TIME : m_intervalValue / 2;
```

Comment:
This value will never change, we can initialize it once

Reply:
It is set to m_intervalValue when the condition is met.